domains/certificates/Query.API/API/DataSyncSyncer/Persistence/SyncState.cs

```diff
@@ -1,33 +1,46 @@
 using System;
-using System.Collections.Generic;
 using System.Linq;
-using API.DataSyncSyncer.Client.Dto;
+using System.Threading.Tasks;
+using API.MasterDataService;
+using API.Query.API.Projections;
 using Marten;
+using Microsoft.Extensions.Logging;
 
 namespace API.DataSyncSyncer.Persistence;
 
 public class SyncState : ISyncState
 {
-    private readonly Dictionary<string, DateTimeOffset> periodStartTimeDictionary = new();
+    private readonly IDocumentStore documentStore;
+    private readonly ILogger<SyncState> logger;
 
-    public void SetNextPeriodStartTime(List<DataSyncDto> measurements, string GSRN)
+    public SyncState(IDocumentStore documentStore, ILogger<SyncState> logger)
     {
-        if (measurements.IsEmpty())
-        {
-            return;
-        }
-
-        var newestMeasurement = measurements.Max(m => m.DateTo);
-        periodStartTimeDictionary[GSRN] = DateTimeOffset.FromUnixTimeSeconds(newestMeasurement);
+        this.documentStore = documentStore;
+        this.logger = logger;
     }
 
-    public long GetPeriodStartTime(string GSRN, DateTimeOffset meteringPointOnboardedStartDate)
+    public async Task<long?> GetPeriodStartTime(MasterData masterData)
     {
-        if (periodStartTimeDictionary.TryGetValue(GSRN, out var periodStart))
+        try
+        {
+            await using var querySession = documentStore.QuerySession();
```

Comment:
I previously had problems with inline using not disposing as expected in https://github.com/Energinet-DataHub/energy-origin/blob/main/domains/auth/Auth.API/API/Utilities/Cryptography.cs#L31

Is this relevant here?

Reply:
It is not the same situation since we are not dealing with streams. Here the using-statement is used to ensure that the session is closed when exiting the method. In Cryptography.cs the using statement would cause all buffered data to be written to the underlying stream when exiting the method; but that would be too late in Cryptography.cs and that is why an explicit call to `.Close()` was added.